client/core/core.go

```diff
@@ -5948,10 +5948,14 @@ func (c *Core) listen(dc *dexConnection) {
 			for _, trade := range doneTrades {
 				// Log an error if redemption funds are still reserved.
 				trade.mtx.RLock()
-				reserved := trade.redemptionLocked
+				redeemLocked := trade.redemptionLocked
+				refundLocked := trade.refundLocked
```

Comment:
Moving the refund unlocking does look more consistent, given we were doing it here for redeems...

This change set does feel better.